scripts/integration-test.sh

```
@@ -0,0 +1,247 @@
+echo -e "${BLUE}║  Preconfirmation Gateway Integration Test Suite       ║${NC}"
+echo -e "${BLUE}╚════════════════════════════════════════════════════════╝${NC}\n"
+
+# Create logs directory
+mkdir -p logs
+
+# cleanup stops mock services and removes the gateway PID file, printing a brief cleanup banner.
+cleanup() {
+  echo -e "\n${BLUE}Cleaning up...${NC}"
+  ./scripts/stop-mock-services.sh
+  if [ -f "logs/gateway.pid" ]; then
+    kill $(cat logs/gateway.pid) 2>/dev/null || true
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)